bittorrent/params.go

```diff
@@ -70,6 +70,17 @@ type RouteParam struct {
 // RouteParams is a collection of RouteParam instances.
 type RouteParams []RouteParam
 
+// ByName returns the value of the first RouteParam that matches the given
+// name. If no matching RouteParam is found, an empty string is returned.
```

Comment:
From the [httprouter README](https://github.com/julienschmidt/httprouter#catch-all-parameters) I see a possible confusion in this scenario: If we use a catch-all parameter like so:
```
/announce/*param
```
and a client requests `/announce/`, then `httprouter` will match the above route and set `param` to `""` (the empty string). Is that correct?
If yes, a consumer of `ByName` can not distinguish between "param not found" and "param set to empty string".

That being said, I think this is expected behaviour (for catch-all parameters). Please add a godoc comment that explains the situation :)